File: src/main/resources/static/entity/subscription.js
export class Subscription{
    id;
    name;
    capacity;
    used;
    remaining;
    price;
    constructor(id, name, capacity, used, remaining, price) {
        this.id = id;
        this.name = name;
        this.capacity = capacity;
        this.used = used;
        this.remaining = remaining;
        this.price = price;
    }
}

File: src/main/java/com/restaurant/restaurantordersystem/entity/Subscription.java
package com.restaurant.restaurantordersystem.entity;

import com.fasterxml.jackson.annotation.JsonIdentityReference;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Data
@Entity
@Table
//@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class , property = "id")
@JsonIdentityReference
public class Subscription {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private Integer capacity;
    private Integer used;
    private Date date;

    public int remaining(){
        return capacity-used;
    }
}
